src/components/input/time-picker/TimePicker.tsx

```diff
@@ -42,87 +54,89 @@ const TimePicker = ({
     }
 
     if (displayEndTime) {
-      // If the end time is displayed, the button should be disabled in the following cases:
-      // 1. Start and end time are the same, and the start time is unchanged
-      // 2. An end time exists and both start and end times are unchanged
       return (
         (areStartAndEndSame && isStartTimeUnchanged) ||
         (endTime && isStartTimeUnchanged && isEndTimeUnchanged)
-      )
+      );
     }
 
-    return false
-  }
+    return false;
+  };
 
   const onButtonClick = (button: ButtonType): void => {
     switch (button) {
       case 'primary':
         // eslint-disable-next-line no-case-declarations
-        let adjustedEndTime: Date | undefined
+        let adjustedEndTime: Date | undefined;
 
         if (displayEndTime && selectedEndTime) {
-          // If the end time is displayed, and the end time is before the start time, add a day to the end time
           if (isBefore(selectedEndTime, selectedStartTime)) {
-            adjustedEndTime = addDays(selectedEndTime, 1)
-            setSelectedEndTime(adjustedEndTime)
+            adjustedEndTime = addDays(selectedEndTime, 1);
+            setSelectedEndTime(adjustedEndTime);
           } else {
-            adjustedEndTime = selectedEndTime
+            adjustedEndTime = selectedEndTime;
           }
         }
 
-        onTimeChange(selectedStartTime, adjustedEndTime)
-        setIsModalOpen(false)
-        break
+        onTimeChange(selectedStartTime, adjustedEndTime);
+        setIsModalOpen(false);
+        break;
       case 'secondary':
-        handleTimeReset()
-        break
+        handleTimeReset();
+        break;
       case 'close':
-        setIsModalOpen(false)
-        setDisplayEndTime(endTime instanceof Date)
-        break
+        setIsModalOpen(false);
+        setDisplayEndTime(endTime instanceof Date);
+        setIs24HourModeTemporary(is24HourModeStored);
+
+        break;
     }
-  }
+  };
 
   const handleTimeReset = (): void => {
-    setSelectedStartTime(startTime)
-    setSelectedEndTime(endTime || startTime)
-  }
+    setSelectedStartTime(startTime);
+    setSelectedEndTime(endTime || startTime);
+  };
 
   const handleEndTimeToggle = (): void => {
     if (displayEndTime) {
-      setTimeout(() => setSelectedEndTime(selectedStartTime), 250)
+      setTimeout(() => setSelectedEndTime(selectedStartTime), 250);
     } else {
-      setSelectedEndTime(endTime || selectedStartTime)
+      setSelectedEndTime(endTime || selectedStartTime);
     }
-    setDisplayEndTime((prev) => !prev)
-  }
-
-  const handleModalOpen = (event: MouseEvent | TouchEvent): void => {
-    event.stopPropagation()
-    const target = event.target as HTMLInputElement
-    target.blur()
+    setDisplayEndTime(prev => !prev);
+  };
+
+  const handle24HourModeTemporaryToggle = (): void => {
+    setIs24HourModeTemporary(prev => !prev);
+    // Optionally, trigger a global toggle here
+    on24HourModeToggle();
```

Comment:
I'm a little confused. You already have the hook which controls the global stored state. Why still keep the props and temporary state? It should be saved immediately to the storage.

Reply:
Because I condition that won't overlaps with 24HourModeState and when I toggled on when in timeMode the 24h mode is turned off it won't save to 24HourModeState and active 24h mode in the settings.

Reply:
Ok, I have done the changes You reuqested, I am uploading the fixed code right now.